refinery/ui/source/js/tool-launch/services/tool-params-service.spec.js

```diff
@@ -0,0 +1,59 @@
+(function () {
+  'use strict';
+
+  describe('Tools Params Service', function () {
+    var mocker;
+    var service;
+
+    beforeEach(module('refineryApp'));
+    beforeEach(module('refineryToolLaunch'));
+    beforeEach(inject(function (mockParamsFactory, toolParamsService) {
+      mocker = mockParamsFactory;
+      service = toolParamsService;
+    }));
+
+    it('service and variables should exist', function () {
+      expect(service).toBeDefined();
+      expect(service.paramsForm).toEqual({});
+      expect(service.toolParams).toEqual([]);
+    });
+
+    it('setSelectedTool is a method', function () {
+      expect(angular.isFunction(service.refreshToolParams)).toBe(true);
+    });
+
+    it('refreshToolParams sets tools param', function () {
+      var tool = {
+        file_relationship: {
+          file_relationship: [],
+          input_files: [],
+          uuid: mocker.generateUuid(),
+          name: 'List of FASTQ Files',
+          value_type: 'LIST'
+        },
+        output_files: [],
+        parameters: [
+          {
+            galaxy_workflow_step: 1,
```

Comment:
I like to mock real-like data, but if we'd like to have a standard let's discuss.